src/lib/orders/index.tsx

```diff
@@ -57,75 +57,112 @@ export function registerOrders(program: Command) {
+      "Filter by maximum filled date (ISO 8601 datestring)",
     )
     .option(
       "--min-fill-price <price>",
-      "Filter by minimum fill price (in cents)"
+      "Filter by minimum fill price (in cents)",
     )
     .option(
       "--max-fill-price <price>",
-      "Filter by maximum fill price (in cents)"
+      "Filter by maximum fill price (in cents)",
     )
     .option("--include-cancelled", "Include cancelled orders")
     .option("--only-cancelled", "Show only cancelled orders")
     .option(
       "--min-cancelled-at <date>",
-      "Filter by minimum cancelled date (ISO 8601 datestring)"
+      "Filter by minimum cancelled date (ISO 8601 datestring)",
     )
     .option(
       "--max-cancelled-at <date>",
-      "Filter by maximum cancelled date (ISO 8601 datestring)"
+      "Filter by maximum cancelled date (ISO 8601 datestring)",
     )
     .option(
       "--min-placed-at <date>",
-      "Filter by minimum placed date (ISO 8601 datestring)"
+      "Filter by minimum placed date (ISO 8601 datestring)",
     )
     .option(
       "--max-placed-at <date>",
-      "Filter by maximum placed date (ISO 8601 datestring)"
+      "Filter by maximum placed date (ISO 8601 datestring)",
     )
     .option("--limit <number>", "Limit the number of results")
     .option("--offset <number>", "Offset the results (for pagination)")
     .option("--json", "Output in JSON format")
-    .action(async options => {
+    .action(async (options) => {
       const minDuration = parseDurationArgument(options.minDuration);
       const maxDuration = parseDurationArgument(options.maxDuration);
 
+      // `--public` implies `--only-open`
+      if (options.public && options.onlyFilled) {
+        logAndQuit(
+          "--public and --only-filled are mutually exclusive. Please choose one.",
+        );
+      }
+      if (options.public && options.onlyCancelled) {
+        logAndQuit(
+          "--public and --only-cancelled are mutually exclusive. Please choose one.",
+        );
+      }
+      if (options.onlyOpen && options.onlyFilled) {
+        logAndQuit(
+          "--only-open and --only-filled are mutually exclusive. Please choose one.",
+        );
+      }
+      if (options.onlyOpen && options.onlyCancelled) {
+        logAndQuit(
+          "--only-open and --only-cancelled are mutually exclusive. Please choose one.",
+        );
+      }
+      if (options.excludeFilled && options.onlyFilled) {
+        logAndQuit(
+          "--exclude-filled and --only-filled are mutually exclusive. Please choose one.",
+        );
+      }
+      if (options.excludeCancelled && options.onlyCancelled) {
+        logAndQuit(
+          "--exclude-cancelled and --only-cancelled are mutually exclusive. Please choose one.",
+        );
+      }
+      if (options.onlyFilled && options.onlyCancelled) {
+        logAndQuit(
+          "--only-filled and --only-cancelled are mutually exclusive. Please choose one.",
+        );
+      }
```

Comment:
You can probably encode this logic using commander

https://github.com/tj/commander.js?tab=readme-ov-file#more-configuration

Reply:
It's a little magical/less explicit than doing it as it is currently, but I think we can lean on commander to parse as much as possible

Reply:
done